### PR TITLE
Allows users to register multiple roles

### LINKED ISSUE #47: Allow users to register multiple roles
`getCurrentUser` should return the current role that the user is signed in to.

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**
Fixes #47. Need to checkout this branch & cherry-pick `local-development`'s commit to test on your local machine.

@chk1994 take note of how I ordered the commits.

@gjlam95 take a look at commit 3's commit message to see whether it makes sense.

@Ley94 take note that `getCurrentUser` returns role in lower case now.

### CODE REVIEW

**Comment by Zhiyuan-Amos** [comment]
Edit: Not done yet, got bug huehue

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -34,7 +34,8 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 
             if (StringUtils.hasText(jwt) && tokenProvider.validateToken(jwt)) {
                 String nric = tokenProvider.getNricFromJWT(jwt);
-                UserDetails userDetails = customUserDetailsService.loadUserByUsername(nric);
+                String role = tokenProvider.getRole(jwt);
```
Comment (gjlam95) [inline backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/security/JwtAuthenticationFilter.java]: I get the commit message but was just wondering if this solves the multiple roles issue? If so, how does this solve it?

**Comment by gjlam95** [comment]
Ohh i get it! I think it's good to go, can I merge this?

**Comment by Zhiyuan-Amos** [comment]
@gjlam95 wait till @chk1994 reviews :P

**Pull request merged**